File: Views/Onboarding/QuestionsIntroStepView.swift
import SwiftUI

struct QuestionsIntroStepView: View {
    @ObservedObject var viewModel: OnboardingViewModel
    
    var body: some View {
        VStack(spacing: 0) {
            // Espace entre barre de progression et titre (harmonisé)
            Spacer()
                .frame(height: 40)
            
            // Titre centré à gauche
            HStack {
                                                              Text("questions_intro_title".localizedOnboarding)
                           .font(.system(size: 28, weight: .bold))
                    .foregroundColor(.black)
                    .multilineTextAlignment(.leading)
                Spacer()
            }
            .padding(.horizontal, 30)
            
            // Espace entre titre et image
            Spacer()
                .frame(height: 40)
            
            // Image de la page de présentation des questions du jour
                    Image("mima") // Image utilisée dans DailyQuestionIntroView
            .resizable()
            .aspectRatio(contentMode: .fit)
            .frame(maxWidth: .infinity, maxHeight: 280)
            .cornerRadius(20)
            .padding(.horizontal, 30)
                .onAppear {
                    print("🖼️ QuestionsIntroStepView: Vue apparue - Affichage de l'image: mima")
                    print("🖼️ QuestionsIntroStepView: Titre de la page: \"questions_intro_title\"")
                }
            
            // Sous-titre descriptif (style paywall)
            VStack(spacing: 16) {
                                       Text("questions_intro_subtitle".localizedOnboarding)
                    .font(.system(size: 16))
                    .foregroundColor(.black.opacity(0.7))
                    .multilineTextAlignment(.center)
                    .lineLimit(nil)
                    .fixedSize(horizontal: false, vertical: true)
                    .padding(.horizontal, 30)
            }
            .padding(.top, 30)
            
            // Deuxième Spacer pour pousser la zone bouton vers le bas
            Spacer()
                
            // Zone blanche collée en bas (style CategoriesPreviewStepView)
            VStack(spacing: 0) {
                Button(action: {
                    print("🔥 QuestionsIntroStepView: Bouton continuer pressé")
                    viewModel.nextStep()
                }) {
                    Text("continue".localized)
                        .font(.system(size: 18, weight: .semibold))
                        .foregroundColor(.white)
                        .frame(maxWidth: .infinity)
                        .frame(height: 56)
                        .background(Color(hex: "#FD267A"))
                        .cornerRadius(28)
                }
                .padding(.horizontal, 30)
            }
            .padding(.vertical, 30)
            .background(Color.white)
            .shadow(color: .black.opacity(0.1), radius: 10, x: 0, y: -5)
        }
        .background(Color(red: 0.97, green: 0.97, blue: 0.98))
    }
}

#Preview {
    QuestionsIntroStepView(viewModel: OnboardingViewModel())
}

File: Views/Onboarding/QuestionsIntroStepView.kt
// QuestionsIntroStepView.kt
package your.package.name

import android.util.Log
import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

@Composable
fun QuestionsIntroStepView(
    viewModel: OnboardingViewModel
) = QuestionsIntroStepView(
    onContinue = {
        Log.d("QuestionsIntroStepView", "Bouton continuer pressé")
        viewModel.nextStep()
    }
)

@Composable
fun QuestionsIntroStepView(
    onContinue: () -> Unit
) {
    val context = LocalContext.current
    val background = Color(0xFFF7F7FA) // équiv. Color(red:0.97, green:0.97, blue:0.98)

    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(background)
    ) {
        Column(modifier = Modifier.fillMaxSize()) {
            Spacer(modifier = Modifier.height(40.dp))

            // Titre aligné à gauche
            Row(
                modifier = Modifier
                    .padding(horizontal = 30.dp)
                    .fillMaxWidth(),
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = stringResource(R.string.questions_intro_title),
                    fontSize = 28.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color.Black,
                    textAlign = TextAlign.Start,
                    modifier = Modifier.weight(1f)
                )
            }

            Spacer(modifier = Modifier.height(40.dp))

            // Image (équiv. Image("mima").resizable().aspectFit...)
            Image(
                painter = painterResource(id = R.drawable.mima),
                contentDescription = null,
                contentScale = ContentScale.Fit,
                modifier = Modifier
                    .padding(horizontal = 30.dp)
                    .fillMaxWidth()
                    .heightIn(max = 280.dp)
                    .clip(RoundedCornerShape(20.dp))
            )

            // Sous-titre style paywall
            Column(
                verticalArrangement = Arrangement.spacedBy(16.dp),
                modifier = Modifier
                    .padding(top = 30.dp)
                    .padding(horizontal = 30.dp)
                    .fillMaxWidth()
            ) {
                Text(
                    text = stringResource(R.string.questions_intro_subtitle),
                    fontSize = 16.sp,
                    color = Color.Black.copy(alpha = 0.7f),
                    textAlign = TextAlign.Center,
                    modifier = Modifier.fillMaxWidth()
                )
            }

            Spacer(modifier = Modifier.weight(1f))

            // Zone blanche collée en bas avec ombre
            Surface(
                color = Color.White,
                tonalElevation = 0.dp,
                shadowElevation = 10.dp,
                modifier = Modifier.fillMaxWidth()
            ) {
                Column(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(vertical = 30.dp)
                ) {
                    Button(
                        onClick = onContinue,
                        shape = RoundedCornerShape(28.dp),
                        colors = ButtonDefaults.buttonColors(
                            containerColor = Color(0xFFFD267A),
                            contentColor = Color.White
                        ),
                        modifier = Modifier
                            .padding(horizontal = 30.dp)
                            .fillMaxWidth()
                            .height(56.dp)
                    ) {
                        Text(
                            text = stringResource(R.string.action_continue),
                            fontSize = 18.sp,
                            fontWeight = FontWeight.SemiBold
                        )
                    }
                }
            }
        }

        // Logs "onAppear"
        LaunchedEffect(Unit) {
            Log.d("QuestionsIntroStepView", "Vue apparue - Affichage de l'image: mima")
            Log.d(
                "QuestionsIntroStepView",
                "Titre de la page: \"${context.getString(R.string.questions_intro_title)}\""
            )
        }
    }
}
